src/test/kotlin/com/example/petshopkotlin/user/AuthControllerTest.kt

```diff
@@ -0,0 +1,151 @@
+package com.example.petshopkotlin.user
+
+import com.example.petshopkotlin.user.model.Role
+import com.example.petshopkotlin.user.model.User
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.http.MediaType
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers.status
+import org.testcontainers.junit.jupiter.Testcontainers
+
+@AutoConfigureMockMvc
+@SpringBootTest
+@Testcontainers
+class AuthControllerTest {
+    @Autowired
+    lateinit var mockMvc: MockMvc
+
+    @Autowired
+    lateinit var userRepository: UserRepository
+
+    @BeforeEach
+    fun setUp() {
+        userRepository.deleteAll()
+    }
+
+    @Test
+    fun `given valid user when register then Success along with userName`() {
+        val content = mockMvc.perform(
+            post("/api/auth")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(
+                    """
+                         {
+                              "userName": "fake_admin123@example.com",
```

Comment:
I would make a method with username, password, and role, that returns a string and use this a template string for the generation of the json.